File: docs/JAMIA/comparisons/RPython/TM/50916422_int64_not_json_serializable.ts
/**
 * ID: SO#50916422
 * Language: Python
 * Bug class: Value type
 * Runtime consequence: Crash
 * In study: Yes
 * Inclusion rationale: numpy int64 extracted from DataFrame not JSON serializable. Tidy-ts values are native JS types.
 */
import { createDataFrame, stats as s } from "@tidy-ts/dataframe";
import { printForeignResult, runForeign } from "../run-foreign.ts";

// Foreign reproduction (pandas) ──────────────────────────────────────────────

const foreignScript = `
import pandas as pd
import json

df = pd.DataFrame({'store': ['A', 'B', 'C'], 'count': [10, 12, 5]})

record = {'name': df['store'].iloc[0], 'count': df['count'].iloc[0]}
json.dumps(record)
`;

printForeignResult("python", runForeign("python", foreignScript));

// Tidy-TS equivalent ─────────────────────────────────────────────────────────

const df = createDataFrame([
  { store: "A", count: 10 },
  { store: "B", count: 12 },
  { store: "C", count: 5 },
]);

// @ts-expect-error — string[] is not assignable to number[]
s.sum(df.extract("store"));

File: docs/JAMIA/comparisons/RPython/TM/50916422_int64_not_json_serializable.py
# ID: SO#50916422
# Language: Python
# Bug class: Value type
# Runtime consequence: Crash
# In study: Yes
# Inclusion rationale: numpy int64 extracted from DataFrame not JSON serializable. Tidy-ts values are native JS types.
# Reproduction status: Live
# Type system catch: Native JS types; no int64/int32 wrapper

import pandas as pd
import json

df = pd.DataFrame({'store': ['A', 'B', 'C'], 'count': [10, 12, 5]})

record = {'name': df['store'].iloc[0], 'count': df['count'].iloc[0]}
print(f"count type: {type(record['count'])}")

try:
    json.dumps(record)
    print("Serialized OK")
except TypeError as e:
    print(f"Crash: {e}")
